Azure/azure-cli
# {Event Grid} Fix TypeError when specifying numerical CLI arguments for certain topic types

**Description**
Commit acd463307edc735b23a3d20fddf2a997f6d746f4 removed the manual casting to int for several cli arguments and fixed that in the argument definition by specifying `type=int` but forgot to do the same for other sets of arguments (system and partner topics).

My GitHub workflow contains a `run` step with the following command:
```
az eventgrid system-topic event-subscription create \
  --name blobeventgrid \
  --resource-group $RESOURCE_GROUP \
  --system-topic-name evto-connect-${{ env.ENVIRONMENT }} \
  --deadletter-endpoint "/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.Storage/storageAccounts/$STORAGEACCOUNTNAME/blobServices/default/containers/XXX" \
  --endpoint "/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.Web/sites/$FUNCTION/functions/XXX" \
  --endpoint-type azurefunction \
  --event-ttl 360 \
  --max-delivery-attempts 10 \
  --included-event-types "Microsoft.Storage.BlobCreated" \
  --subject-ends-with ".xml" \
  --labels "github-actions"
```
Until now this always worked as expected. GitHubs base image for workflows recently updated to Azure CLI v2.22.0 and since then I got:
```
ERROR: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 657, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 720, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 691, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 807, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/eventgrid/custom.py", line 683, in cli_system_topic_event_subscription_create_or_update
    delivery_attribute_mapping=delivery_attribute_mapping)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/eventgrid/custom.py", line 952, in _get_event_subscription_info
    _validate_retry_policy(max_delivery_attempts, event_ttl)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/eventgrid/custom.py", line 1547, in _validate_retry_policy
    if max_delivery_attempts < 1 or max_delivery_attempts > 30:
TypeError: '<' not supported between instances of 'str' and 'int'
```

**Testing Guide**
Just enter any `az eventgrid system-topic event-subscription create` command while specifying one of `max-delivery-attempts`, `max-events-per-batch`, `preferred-batch-size-in-kilobytes` or `event-ttl`.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
